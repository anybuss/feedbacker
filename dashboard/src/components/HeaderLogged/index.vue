<template>
  <section>
    <div class="wrapper">
      <img src="@/assets/images/logo_white.svg" alt="Feedbacker logo" />

      <nav>
        <ul>
          <li>Credenciais</li>
          <li>Feedbacks</li>
          <li>Sair</li>
        </ul>
      </nav>
    </div>
  </section>
</template>

<style lang="scss" scoped>
section {
  background-color: $main-color;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 112px;
  .wrapper {
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 80%;
    max-width: 1200px;
    margin: 0 auto;
    padding: 2rem 0;
    img {
      width: 150px;
    }
    nav {
      ul {
        display: flex;
        align-items: center;
        gap: 2.5rem;
        list-style: none;
        li {
          color: $white-color;
          font-weight: $font-bold;
          cursor: pointer;
          &:focus {
            outline: none;
          }
          &:last-child {
            @include button-primary;
          }
        }
      }
    }
  }
}

@media screen and (max-width: 780px) {
  section {
    height: auto;
    .wrapper {
      flex-direction: column;
      align-items: center;
      nav {
        margin-top: 2rem;
        ul {
          flex-direction: column;
          gap: 1rem;
          li {
            &:last-child {
              margin-top: 1.5rem;
            }
          }
        }
      }
    }
  }
}
</style>
