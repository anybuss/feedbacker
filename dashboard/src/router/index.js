import { createRouter, createWebHistory } from "vue-router";

const Home = () => import("../views/Home/index.vue");
const Feedbacks = () => import("../views/Feedbacks/index.vue");

const routes = [
  {
    path: "/",
    name: "Home",
    component: Home,
  },
  {
    path: "/",
    name: "Feedbacks",
    component: Feedbacks,
  },
  {
    path: "/:pathMatch(.*)*",
    redirect: { name: "Home" },
  },
];

const router = createRouter({
  history: createWebHistory("/"),
  routes,
});

export default router;
