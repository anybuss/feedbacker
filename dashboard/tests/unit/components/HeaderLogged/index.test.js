import HeaderLogged from "@/components/HeaderLogged/index.vue";
import { routes } from "@/router/index";
import { createRouter, createWebHistory } from "vue-router";
import { shallowMount } from "@vue/test-utils";

const router = createRouter({
  history: createWebHistory("/"),
  routes,
});

describe("HeaderLogged component", () => {
  it("should render header logged correctly", async () => {
    router.push("/");
    await router.isReady();
    const wrapper = shallowMount(HeaderLogged, {
      global: {
        plugins: [router],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render 3 dots when user isn't logged", async () => {
    router.push("/");
    await router.isReady();
    const wrapper = shallowMount(HeaderLogged, {
      global: {
        plugins: [router],
      },
    });

    const buttonLogout = wrapper.find("#navbar-logout");
    expect(buttonLogout.text()).toBe("...");
  });
});
