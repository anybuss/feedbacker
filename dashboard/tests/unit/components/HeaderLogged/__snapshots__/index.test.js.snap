// Vitest Snapshot v1

exports[`HeaderLogged component > should render header logged correctly 1`] = `
"<section data-v-dfaab776=\\"\\">
  <div data-v-dfaab776=\\"\\" class=\\"wrapper\\"><img data-v-dfaab776=\\"\\" src=\\"/src/assets/images/logo_white.svg\\" alt=\\"Feedbacker logo\\">
    <nav data-v-dfaab776=\\"\\">
      <ul data-v-dfaab776=\\"\\">
        <li data-v-dfaab776=\\"\\" id=\\"navbar-credentials\\"> Credenciais </li>
        <li data-v-dfaab776=\\"\\" id=\\"navbar-feedbacks\\">Feedbacks</li>
        <li data-v-dfaab776=\\"\\" id=\\"navbar-logout\\">...</li>
      </ul>
    </nav>
  </div>
</section>"
`;
